step-hello-world.go

```diff
@@ -45,10 +45,16 @@ func printUsage() {
 func StateMachine() (*machine.StateMachine, error) {
 	state_machine, err := machine.FromJSON([]byte(`{
     "Comment": "Hello World",
-    "StartAt": "Hello",
+    "StartAt": "HelloFn",
     "States": {
+      "HelloFn": {
+        "Type": "Pass",
+        "Result": "Hello",
+        "ResultPath": "$.Task",
+        "Next": "Hello"
+      },
       "Hello": {
-        "Type": "TaskFn",
+        "Type": "Task",
```

Comment:
You should use TaskFn type. These are custom step types that add the Pass state automatically, to reduce the massive amount of boiler plate they add and force naming standards.

Reply:
The problem is that the state that is produced by TaskFn, `_hello_`, seems to break the state machine when deployed to AWS – it fails with Cannoty Find Task errors:

```
{
            "timestamp": 1538598688.651,
            "type": "LambdaFunctionFailed",
            "id": 7,
            "previousEventId": 6,
            "lambdaFunctionFailedEventDetails": {
                "error": "TaskError",
                "cause": "{\"errorMessage\":\"TaskError(Hello) : [_Hello_]: Cannot Find Task\",\"errorType\":\"TaskError\"}"
            }
        },
        {
            "timestamp": 1538598688.651,
            "type": "ExecutionFailed",
            "id": 8,
            "previousEventId": 7,
            "executionFailedEventDetails": {
                "error": "TaskError",
                "cause": "{\"errorMessage\":\"TaskError(Hello) : [_Hello_]: Cannot Find Task\",\"errorType\":\"TaskError\"}"
```

Reply:
Here's the definition the TaskFn version produced
```
{
 "Comment": "Hello World",
 "StartAt": "Hello",
 "States": {
  "Hello": {
   "Next": "_Hello_",
   "Result": "Hello",
   "ResultPath": "$.Task",
   "Type": "Pass"
  },
  "_Hello_": {
   "Comment": "Deploy Step Function",
   "End": true,
   "Type": "Task"
  }
 }
}
```

It seems like it's failing because the `Result` produced is `Hello` rather than `_Hello_` as it's working when I manually change that in AWS

Reply:
I think these issues occur because I have not done the best job keeping `step-hello-world` up to date with updates in `step`. #6 Addresses this and updates step, and the tests, and this should now work.